src/Splitter.sol

```diff
@@ -0,0 +1,113 @@
+// SPDX-FileCopyrightText: © 2023 Dai Foundation <www.daifoundation.org>
+// SPDX-License-Identifier: AGPL-3.0-or-later
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.8.16;
+
+interface VatLike {
+    function move(address, address, uint256) external;
+    function hope(address) external;
+    function nope(address) external;
+}
+
+interface DaiJoinLike {
+    function vat() external view returns (address);
+    function exit(address, uint256) external;
+}
+
+interface FlapLike {
+    function kick(uint256, uint256) external returns (uint256);
+    function cage(uint256) external;
+}
+
+interface FarmLike {
+    function notifyRewardAmount(uint256 reward) external;
+}
+
+contract Splitter {
+    mapping (address => uint256) public wards;
+
+    VatLike     public immutable vat;
+    DaiJoinLike public immutable daiJoin;
+    FarmLike    public immutable farm;
+
+    FlapLike public flapper;
```

Comment:
nit - I think we lately tend to put all the storage vars together.

Reply:
Changed

Reply:
Oh yes, done.